apis/v1alpha1/shared_types.go

```diff
@@ -80,77 +87,53 @@ type Port struct {
-// option has been specified and fail closed.
+// AdminNetworkPolicyIngressPeer defines a peer to allow traffic to.
+//
+// Exactly one of the fields must be set for a given peer and this is enforced
+// by the validation rules on the CRD. If an implementation sees no fields are
+// set then it can infer that the deployed CRD is of an incompatible version
+// with an unknown field.  In that case it should fail closed.
+//
+// For "Allow" rules, "fail closed" means: "treat the rule as matching no
+// traffic". For "Deny" and "Pass" rules, "fail closed" means: "treat the rule
+// as a 'Deny all' rule".
+//
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
-type AdminNetworkPolicyPeer struct {
-	// Namespaces defines a way to select a set of Namespaces.
+type AdminNetworkPolicyIngressPeer struct {
+	// Namespaces defines a way to select all pods within a set of Namespaces.
+	// Note that host-networked pods are not included in this type of peer.
+	//
 	// +optional
-	Namespaces *NamespacedPeer `json:"namespaces,omitempty"`
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
 	// Pods defines a way to select a set of pods in
-	// in a set of namespaces.
+	// a set of namespaces. Note that host-networked pods
+	// are not included in this type of peer.
+	//
 	// +optional
-	Pods *NamespacedPodPeer `json:"pods,omitempty"`
+	Pods *NamespacedPod `json:"pods,omitempty"`
 }
 
-// NamespacedPeer defines a flexible way to select Namespaces in a cluster.
-// Exactly one of the selectors must be set.  If a consumer observes none of
-// its fields are set, they must assume an unknown option has been specified
-// and fail closed.
-// +kubebuilder:validation:MaxProperties=1
-// +kubebuilder:validation:MinProperties=1
-type NamespacedPeer struct {
-	// NamespaceSelector is a labelSelector used to select Namespaces, This field
-	// follows standard label selector semantics; if present but empty, it selects
-	// all Namespaces.
-	// +optional
-	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
-
-	// SameLabels is used to select a set of Namespaces that share the same values
-	// for a set of labels.
-	// To be selected a Namespace must have all of the labels defined in SameLabels,
-	// AND they must all have the same value as the subject of this policy.
-	// If Samelabels is Empty then nothing is selected.
-	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	SameLabels []string `json:"sameLabels,omitempty"`
-
-	// NotSameLabels is used to select a set of Namespaces that do not have certain
-	// values for a set of label(s).
-	// To be selected a Namespace must have all of the labels defined in NotSameLabels,
-	// AND at least one of them must have different values than the subject of this policy.
-	// If NotSameLabels is empty then nothing is selected.
-	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	NotSameLabels []string `json:"notSameLabels,omitempty"`
```

Comment:
We removed `SameLabels`/`NotSameLabels` because of problems with the version of "tenancy" that they implemented (https://github.com/kubernetes-sigs/network-policy-api/issues/122). We still have not re-added tenancy.